hardware/detect.py

```diff
@@ -709,6 +709,11 @@ def _get_governor(lcpu):
             lscpux[item.strip(':')] = value.strip()
 
     hw_lst.append(("cpu", "physical", "number", int(lscpu["Socket(s)"])))
+
+    with contextlib.suppress(IOError):
+        value = _from_file("/sys/devices/system/cpu/smt/control")
```

Comment:
Don't you want active instead of control ?

Reply:
They have a different output, but they can both be used to read the SMT state. "active" has values 0 and 1, "control" has values "on" "off" "forceoff" "notsupported" "notimplemented".

Following some tests we found out that, in some cases, there might be a discrepancy between the value in "active" and "control", with active with value 0 or off and control with value on. This happens when SMT is enabled but the siblings are offline.

Given this discrepancy and the fact that both this measures don't extract the number of threads per core, I suggest to expose instead the field "Thread(s) per core" of lscpu. Would you agree to this change?

Reply:
I didn't tested this as much as it seems you did so I'll trust your testing. Let's use control then.
Yeah, reported threads per core is a good idea.

Reply:
It would be a good idea to have some tests for the smt value.

Reply:
OK, I'll implement both threads per core and SMT using "control".

What do you have in mind as tests for the smt value?

Reply:
Something like the boost testing in test_get_cpus_vm(). That's very simple.

Reply:
Are you talking about

`     for i in range(1):
        calls.append(mock.call('/sys/devices/system/cpu/cpufreq/boost'))`
?

This kind of test is already there for `smt` too.